gateway/handler_ipns_record.go

```diff
@@ -59,7 +59,7 @@ func (i *handler) serveIpnsRecord(ctx context.Context, w http.ResponseWriter, r
 	// TTL is not present, we use the Last-Modified tag. We are tracking IPNS
 	// caching on: https://github.com/ipfs/kubo/issues/1818.
 	// TODO: use addCacheControlHeaders once #1818 is fixed.
-	w.Header().Set("Etag", getEtag(r, resolvedPath.Cid()))
+	w.Header().Set("Etag", getEtag(r, c))
```

Comment:
:warning:  We should set `Etag` to a hash based on `rawRecord` bytes.

We already use xxhash for `AssetHash` in `getDirListingEtag`, so maybe something like:

```suggestion
    recordEtag := strconv.FormatUint(xxhash.Sum64(rawRecord), 32)
	w.Header().Set("Etag", recordEtag)
```

(if this breaks sharness, fine to fix in separate PR – just fill issue so we dont forget to fix)